UIImageColors/Sources/UIImageColors.swift

```diff
@@ -247,7 +247,7 @@ extension UIImage {
         let imageColors = NSCountedSet(capacity: width*height)
         for x in 0..<width {
```

Comment:
Wouldn't this also affect the `x` range?

Reply:
No. The image’s width does not change.

Reply:
I'm not sure I understand. We are determining the pixel index based on the `bytesPerRow` by iterating over the width range. If they deviate, will this not cause an index out of bounds exception on `data`?

Reply:
No, this will not cause an out of bounds exception. `bytesPerRow` is always greater or equal to `width * bytesPerComponent * componentCount` (`width * 1 * 4` in our case).

Reply:
Sorry for going back and forth on this, but I'm still finding it hard to understand why width and `bytesPerRow` can't deviate — considering that is the assumption we are making to index the value in the data array. Is it possible to provide an example image I can test?

Reply:
I created a SwiftUI app that you can use to run some tests: https://gist.github.com/YuAo/e28e568184c413ffb5acf9374043e377

This app creates a 30x30 image with a pink dot in the center. It provides two different way of enumerating pixels in the image, using `width` or using `bytesPerRow`.

<img width="399" alt="Screen Shot 2021-04-04 at 12 20 19" src="https://user-images.githubusercontent.com/1234944/113498573-bd38e480-9540-11eb-83fc-f8357bd8c94d.png">
<img width="413" alt="Screen Shot 2021-04-04 at 12 20 29" src="https://user-images.githubusercontent.com/1234944/113498574-bf9b3e80-9540-11eb-8bc4-97367b1d99bd.png">

Reply:
Awesome, thanks for the example. LGTM!